fc-examine/lib/examine.go

```diff
@@ -0,0 +1,450 @@
+	"bytes"
+	"context"
+	"fmt"
+	"math/big"
+	"reflect"
+	"regexp"
+	"strings"
+
+	addr "github.com/filecoin-project/go-address"
+	bitfield "github.com/filecoin-project/go-bitfield"
+	chainState "github.com/filecoin-project/lotus/chain/state"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/lib/blockstore"
+	"github.com/filecoin-project/specs-actors/actors/builtin"
+	accountActor "github.com/filecoin-project/specs-actors/actors/builtin/account"
+	cronActor "github.com/filecoin-project/specs-actors/actors/builtin/cron"
+	initActor "github.com/filecoin-project/specs-actors/actors/builtin/init"
+	marketActor "github.com/filecoin-project/specs-actors/actors/builtin/market"
+	storageMinerActor "github.com/filecoin-project/specs-actors/actors/builtin/miner"
+	storagePowerActor "github.com/filecoin-project/specs-actors/actors/builtin/power"
+	rewardActor "github.com/filecoin-project/specs-actors/actors/builtin/reward"
+	verifiedRegistryActor "github.com/filecoin-project/specs-actors/actors/builtin/verifreg"
+	runtime "github.com/filecoin-project/specs-actors/actors/runtime"
+	adt "github.com/filecoin-project/specs-actors/actors/util/adt"
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	hamt "github.com/ipfs/go-hamt-ipld"
+	cbor "github.com/ipfs/go-ipld-cbor"
+	cbg "github.com/whyrusleeping/cbor-gen"
+	"github.com/willscott/go-cmp/cmp"
+)
+
+type config struct {
+	ExpandActors   bool
+	ActorCidFilter []cid.Cid
+}
+
+type Option func(c *config)
+
+func ExpandActors(c *config) {
+	c.ExpandActors = true
+}
+
+func ExpandActorByCid(cids []cid.Cid) Option {
+	return func(c *config) {
+		c.ExpandActors = true
+		c.ActorCidFilter = cids
+	}
+}
+
+// Parse a user entered fuzzy definition for actor expansion.
+func WithActorExpansionFromUser(arg string) (Option, error) {
+	if arg == "all" {
+		return ExpandActors, nil
+	}
+	parts := strings.Split(arg, ",")
+
+	cids := make([]cid.Cid, 0, len(parts))
+	for _, part := range parts {
+		c, err := cid.Parse(part)
+		if err != nil {
+			for k, v := range builtinCodeClasses {
+				if part == k {
+					cids = append(cids, v)
+					break
+				}
+			}
+			return nil, fmt.Errorf("Could not parse %s: %w", part, err)
+		} else {
+			cids = append(cids, c)
+		}
+	}
+	return ExpandActorByCid(cids), nil
+}
+
+var builtinCodeClasses = map[string]cid.Cid{
+	"init":             builtin.InitActorCodeID,
+	"cron":             builtin.CronActorCodeID,
+	"account":          builtin.AccountActorCodeID,
+	"storagePower":     builtin.StoragePowerActorCodeID,
+	"storageMiner":     builtin.StorageMinerActorCodeID,
+	"storageMarket":    builtin.StorageMarketActorCodeID,
+	"paymentChannel":   builtin.PaymentChannelActorCodeID,
+	"multisig":         builtin.MultisigActorCodeID,
+	"reward":           builtin.RewardActorCodeID,
+	"verifiedRegistry": builtin.VerifiedRegistryActorCodeID,
+}
+
+func Diff(ctx context.Context, store blockstore.Blockstore, a, b cid.Cid, opts ...Option) {
```

Comment:
Having this print straight to stdout is a bit inconvenient. Can we make it return a string? Or provide an alternate `Sdiff` that returns a `string`?